delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/ResourceF.scala

```diff
@@ -84,9 +88,44 @@ object ResourceF {
       r.void.asJsonObject deepMerge r.value.asJsonObject
     }
 
+  private def resourceFUnitJsonLdEncoder(
+      valueContext: ContextValue
+  )(implicit base: BaseUri): JsonLdEncoder[ResourceF[Unit]] =
+    JsonLdEncoder.computeFromCirce(
+      _.id.resolvedAgainst(base.endpoint.toIri),
+      ContextValue(contexts.metadata).merge(valueContext)
+    )
+
   implicit final def resourceFUnitJsonLdEncoder(implicit base: BaseUri): JsonLdEncoder[ResourceF[Unit]] =
-    JsonLdEncoder.computeFromCirce(_.id.resolvedAgainst(base.endpoint.toIri), ContextValue(contexts.metadata))
+    resourceFUnitJsonLdEncoder(ContextValue.empty)
+
+  implicit def resourceFAJsonLdEncoder[A](implicit
```

Comment:
why are you reimplementing the `compose ` logic customized here? Why not to do it directly in `JsonLdEncoder.compose` ?

Reply:
Because we can't create new JsonLdEncoders to create new ones like in your recommendation as they keep using the previous context and not the merged ones so it did not fix the bug.
It is the only place where compose is used in the code.

Reply:
Then there must be something else to be fixed. As far as I can see, most of the implementations of `JsonLdEncoder` in on the core and plugins codebase use `JsonLdEncoder.computeFromCirce`. In this case it should work, because when [compacting we do rely on the call to the context method](https://github.com/BlueBrain/nexus/blob/master/delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/jsonld/encoder/JsonLdEncoder.scala#L152).

When changing the JsonLdEncoder on `compose` overriding the `context` method, should affect all those JsonLdEncoder instances built with `computeFromCirce`.

Reply:
for those cases, overriding the JsonLdContext in the `compose` function with something on the following lines should work:
```scala
...
val AA = A.withContext(context(value))
val BB = B.withContext(context(value))
...
```

Reply:
No because the self.compact calls the self.context and not the merge one we introduce.
And not all depend on Circe so we can't generalize it (Schemas and Resources for example).
So far only ResourceF is concerned.

Reply:
And apart from this cornercase, JsonLdEncoder is working fine so we don't need to change its interface to address this issue (and less on a branch supposed to fix a bug)

Reply:
`Schemas` and `Resources ` are exceptions and do not need that anyway, so it isn't a concern.

For the rest, I'm gonna do some checks myself and let you know.